### DISCUSSION TITLE
How to save settings from Property Inspector using new SDK

### DISCUSSION THREAD

**Discussion by brandonenr7** [elgatosf/elgato#123] [Help] [answered]
So I'm using the new [SDK](https://github.com/elgatosf/streamdeck) and have gotten the idea of how to create the actions, I even was able to define in the manifest within the property inspector html for the action and it displayers properly. Using the previous method (javascript) we'd inject the inspector.js file and define functions, etc. like how the current docs show. However with the new SDK, how do we get those settings from the PI and store them into the action settings?

I won't lie, I am a bit newer to TypeScript so maybe I'm overlooking something. Also is there any docs on the new SDK yet as I've been just looking through the source code.

Update: Figured out a way [using the docs](https://docs.elgato.com/sdk/plugins/property-inspector#sample-property-inspector) for the Javascript way, I was wondering if there was just more of a Node/TS method to doing it.

**Comment by ChekTek** [comment, marked as answer]
Hi @brandonenr7 

The docs and property inspector are a work in progress at the moment. We'll share a more complete post here on [GitHub](https://github.com/orgs/elgatosf/discussions) as well as in the [Maker Discord](https://discord.gg/Nv7Qtjkcs8) when things are ready. For now, I'd recommend taking a look at https://sdpi-components.dev/ as we have plans to implement something very similar
